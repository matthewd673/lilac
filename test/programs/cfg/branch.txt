func main() -> void
  i32 $a = 5i32
  jnz $a L2

  L1:
  i32 $a = 3i32
  jmp L3

  L2:
  i32 $a = 4i32

  L3:
  i32 $b = $a * 2i32
end
