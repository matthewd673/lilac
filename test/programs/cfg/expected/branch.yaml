---
nodes:
- ENTRY
- BB_1
- BB_2
- BB_3
- BB_4
- EXIT
edges:
- from: ENTRY
  to: BB_1
- from: BB_1
  to: BB_2
- from: BB_1
  to: BB_3
- from: BB_2
  to: BB_4
- from: BB_3
  to: BB_4
- from: BB_4
  to: EXIT
